dockstore-webservice/src/main/java/io/dockstore/webservice/core/Tag.java

```diff
@@ -198,18 +215,19 @@ public boolean equals(Object obj) {
             return false;
         }
         final Tag other = (Tag)obj;
-        return Objects.equals(this.getId(), other.getId());
+        return Objects.equals(this.getId(), other.getId()) && Objects.equals(this.lastBuilt, other.lastBuilt);
```

Comment:
I don't think this is right, although it looks like it was already like this.

See [this article](https://stackoverflow.com/questions/17027777/relationship-between-hashcode-and-equals-method-in-java), where it says `The part of the contract here which is important is: objects which are .equals() MUST have the same .hashCode().`

In practice, this may not matter, because the ID is unique. If it doesn't matter, then we should probably change the `hashCode` method to only has the id.

Unless there is some Hibernate complication I'm missing...

Reply:
(Hmmm, I think both the original and the suggested fix might be wrong. https://docs.jboss.org/hibernate/core/4.0/manual/en-US/html/persistent-classes.html#persistent-classes-equalshashcode )

I think for now, continue the pattern. Create an issue to investigate whether we're affected by the issue described.